crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -287,14 +371,79 @@ impl RegularPolygon {
     ///
     /// # Panics
     ///
-    /// Panics if `circumcircle_radius` is non-positive
-    pub fn new(circumcircle_radius: f32, sides: usize) -> Self {
-        assert!(circumcircle_radius > 0.0);
+    /// Panics if `circumradius` is non-positive
+    pub fn new(circumradius: f32, sides: usize) -> Self {
+        assert!(circumradius > 0.0, "polygon has a non-positive radius");
+        assert!(sides > 2, "polygon has less than 3 sides");
+
         Self {
             circumcircle: Circle {
-                radius: circumcircle_radius,
+                radius: circumradius,
             },
             sides,
         }
     }
+
+    /// Get the radius of the circumcircle on which all vertices
+    /// of the regular polygon lie
+    pub fn circumradius(&self) -> f32 {
+        self.circumcircle.radius
+    }
+
+    /// Get the apothem or inradius of the regular polygon.
+    /// This is the radius of the largest circle that can
+    /// be drawn within the polygon
+    #[doc(alias = "inradius")]
+    pub fn apothem(&self) -> f32 {
+        self.circumradius() * (PI / self.sides as f32).cos()
+    }
+
+    /// Get the length of one side of the regular polygon
+    pub fn side_length(&self) -> f32 {
+        2.0 * self.circumradius() * (PI / self.sides as f32).sin()
+    }
+
+    /// Get the area of the regular polygon
+    pub fn area(&self) -> f32 {
+        let apothem = self.apothem();
+        self.sides as f32 * apothem * apothem * (PI / self.sides as f32).tan()
+    }
+
+    /// Get the perimeter of the regular polygon.
+    /// This is the sum of its sides
+    pub fn perimeter(&self) -> f32 {
+        self.sides as f32 * self.side_length()
+    }
+
+    /// Get the internal angle of the regular polygon in degrees.
+    ///
+    /// This is the angle formed by two adjacent sides with points
+    /// within the angle being in the interior of the polygon
+    pub fn internal_angle_degrees(&self) -> f32 {
```

Comment:
Is it fine to have separate `_degrees` and `_radians` versions, or should there be just one, possibly with degrees? Using separate versions makes the computation slightly more efficient since the user doesn't need to convert the units.